django/db/models/query_utils.py

```diff
@@ -46,10 +45,10 @@ def _combine(self, other, conn):
 
         # If the other Q() is empty, ignore it and just use `self`.
         if not other:
-            return copy.deepcopy(self)
+            return self
         # Or if this Q is empty, ignore it and just use `other`.
         elif not self:
-            return copy.deepcopy(other)
+            return other
```

Comment:
That won't work, `_combine` must return a _new_ object.

I suggest we adapt the code to catch `TypeError` on `deepcopy` and use the old code path on failure.

Reply:
Thanks for your review, I think if catching the exception and using the old code does not meet our needs, can I use deconstruct to recreate an object?